### ISSUE TITLE
V2: CV, multicomponent dataset size bug

### ISSUE THREAD

**Issue by KnathanM** [chemprop/chemprop#646]
Data split sizes [log](https://github.com/chemprop/chemprop/blob/v2/dev/chemprop/cli/train.py#L563) as [1, 1, 1] when using cross validation.
I think we need to be more careful with the structure of our datasets to keep it general because a set of data can have multiple (1) folds of cross validation, (2) molecules per data point, and (3) data points. Are there any other "multiples" in a data structure that we need to worry about?

### Data structure of splits
The most basic data for training is a list of SMILES strings. If have multiple components for a data point we would have a list of component lists. If we are doing cross validation, we would have a list of fold lists of component lists. 

As an example, suppose my data has three data points of reactions in solvent: (R1, S1), (R2, S2), (R3, S3)

My training data would look like this if I had three folds.
[[[R1, R2],[S1, S2]],
 [[R2, R3],[S2, S3]],
 [[R1, R3],[S1, S3]]]
 
If I was only doing one fold, the data structure should still have an extra list nesting to signify it is only one fold.
[[[R1, R2],[S1, S2]]]

If I actually didn't care about the solvents, the data structure should still have an extra list nesting to signify only one component.
[[[R1, R2]],
 [[R2, R3]],
 [[R1, R3]]]
 
A simple one fold, one component data structure would then be a list in a list in a list. That is a bit annoying but it keeps the data structure consistent.
[[[R1, R2]]]

In summary:
data[0] is fold 0
data[0][0] is all R
data[0][0][0] is R1

The code seems to be easier if this structure is preserved for as long as possible. I propose `build_splits()` should always produce a tuple of three lists, where each list corresponds to train, val, or test and contains `n` lists corresponding to the number of folds. Each of those `n` lists contains `m` lists corresponding to the number of input molecules per data point. Each of those `m` lists contains `d` elements that are the set of inputs to train the model on. The `for` loop in `main()` would loop through the `n` lists. Then `build_datasets()` would handle the `m` molecules per data point.

### Building data splits
`build_splits()` builds the splits by first calling `build_data_from_files()` and then optionally by calling `split_component()`. So both of these functions need to return their data in the standard data structure and `split_component()` also needs to accept data in this standard data structure.
(Note: Cross validation doesn't make sense with supplying your own validation and test set, so if cross validation is used and a separate data set is given, an error should be thown.)

Opening an issue here for discussion before I push changes to a PR.
